### PR TITLE
JNI: Push back decimal utils from spark-rapids

### PR DESCRIPTION
**Pull request by sperlingxx**
Current PR is to push back cuDF-related decimal utilities from spark-rapids (for https://github.com/NVIDIA/spark-rapids/issues/3793).

These utils were manually verified through spark-rapids integration tests.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/DType.java [hunk 1]
```diff
@@ -173,6 +173,38 @@ private DType(DTypeEnum id, int decimalScale) {
...
+    return DType.DECIMAL128_MAX_PRECISION;
+  }
+
+  /**
+   * Get the number of decimal places needed to hold the Integral Type.
+   * NOTE: this method is NOT for Decimal Type but for Integral Type
+   *
+   * @return the minimum decimal precision (places) for Integral Type
+   */
+  public int getPrecisionForIntegralType() {
```
Comment (ttnghia) [inline java/src/main/java/ai/rapids/cudf/DType.java:195]:
```suggestion
  public int getPrecisionDigits() {
```

Comment (sperlingxx) [reply java/src/main/java/ai/rapids/cudf/DType.java]: I renamed it as `getPrecisionForInt`.

**Comment by sperlingxx** [comment]
rerun tests

**Comment by ttnghia** [comment]
Hey, this is forgotten.

**Review by ttnghia — APPROVED** [review]
Approve but approval is not counted because I'm not java-codeowner :smiley:

**Comment by sperlingxx** [comment]
@gpucibot merge

**Pull request merged**